presto-influx/src/test/java/io/prestosql/plugin/influx/InfluxQueryRunner.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.influx;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logger;
+import io.airlift.tpch.TpchTable;
+import io.prestosql.Session;
+import io.prestosql.metadata.QualifiedObjectName;
+import io.prestosql.plugin.tpch.TpchPlugin;
+import io.prestosql.spi.security.Identity;
+import io.prestosql.spi.security.SelectedRole;
+import io.prestosql.testing.DistributedQueryRunner;
+import io.prestosql.testing.MaterializedResult;
+import io.prestosql.testing.MaterializedRow;
+import io.prestosql.testing.assertions.Assert;
+import okhttp3.Credentials;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import org.testcontainers.containers.InfluxDBContainer;
+
+import java.io.Closeable;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static io.airlift.units.Duration.nanosSince;
+import static io.prestosql.plugin.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.prestosql.spi.security.SelectedRole.Type.ROLE;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static java.util.Locale.ENGLISH;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.testcontainers.containers.InfluxDBContainer.INFLUXDB_PORT;
+
+public class InfluxQueryRunner
+        implements Closeable
+{
+    public static final String DATABASE = InfluxTpchTestSupport.TEST_DATABASE;
+    public static final String USERNAME = "testuser";
+    public static final String PASSWORD = "testpass";
+    public static final String ADMIN_USERNAME = "admin";
+    public static final String ADMIN_PASSWORD = "root";
+    private static final String CATALOG = "influx_tests";
+    public static final String TPCH_SCHEMA = InfluxTpchTestSupport.TPCH_SCHEMA;
+    private final Logger logger = Logger.get("InfluxQueryRunner");
+    private final InfluxDBContainer dockerContainer;
+    private final DistributedQueryRunner queryRunner;
+    private boolean initedTpch;
+
+    public InfluxQueryRunner()
+            throws Exception
+    {
+        dockerContainer = new InfluxDBContainer()
```

Comment:
Please extract `dockerContainer` from this class. You can refer to `TestingMySqlServer` class.